packages/core/src/Provider/HandleProvider/types.ts

```diff
@@ -0,0 +1,30 @@
+import { Cardano, Point } from '../..';
+
+export type Handle = string;
+
+export type CardanoAddress =
+  | Cardano.ByronAddress
+  | Cardano.PaymentAddress
+  | Cardano.EnterpriseAddress
+  | Cardano.PointerAddress;
+
+export enum HandleIssuer {
+  KoraLabs = 'KoraLabs'
+}
+export interface HandleResolution {
+  policyId: Cardano.PolicyId;
+  handle: Handle;
+  hasDatum: boolean;
+  resolvedAddresses: {
+    cardano: CardanoAddress;
+  };
+  resolvedAt: Point;
+}
+
+export interface ResolveHandlesArgs {
+  handles: Handle[];
+}
+
+export interface HandleProvider {
```

Comment:
```suggestion
export interface StakePoolProvider extends Provider {
```

This is how usually providers are defined in order to let them inherit the `healthCheck()` method.
Next it must be implemented as well, but I see the `KoraLabsHandleProvider` already correctly implements it.

Reply:
dde35b14797987 should address this